tests/test_solvers/test_harbor_solver.py

```
@@ -386,6 +387,164 @@ def test_exposes_all_base_environment_public_attrs(self, tmp_path):
+
+
+class TestPatchedTokenCounter:
+    """litellm.utils.token_counter is wrapped to inject the registered custom tokenizer."""
+
+    @pytest.fixture(autouse=True)
+    def _reset_state(self, monkeypatch):
+        monkeypatch.setattr(harbor_mod, "_TOKEN_COUNTER_PATCHED", False)
+        monkeypatch.setattr(harbor_mod, "_TOKENIZER_REGISTRY", {})
+
+    @staticmethod
+    def _install_recorder(monkeypatch):
```

Ruff (0.15.18)

[warning] Missing return type annotation for staticmethod `_install_recorder`

(ANN205)